storagenode/piecestore/retain.go

```diff
@@ -0,0 +1,186 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package piecestore
+
+import (
+	"context"
+	"runtime"
+	"sync"
+	"time"
+
+	"go.uber.org/zap"
+
+	"storj.io/storj/pkg/bloomfilter"
+	"storj.io/storj/pkg/storj"
+	"storj.io/storj/storagenode/pieces"
+)
+
+// RetainService queues and processes retain requests from satellites
+type RetainService struct {
```

Comment:
Is there a reason it's not a separate package?

Reply:
No. I will move it to another package.

Reply:
done